File: tests/src/test/java/issues/issue1.java
package issues;

import clojure.lang.Keyword;
import datomicClient.anomaly.CognitectAnomaly;
import datomicJava.SetupAsync;
import datomicJava.client.api.async.*;
import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;
import java.util.stream.Stream;

import static datomic.Util.*;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

@FixMethodOrder(MethodSorters.JVM)
public class issue1 extends SetupAsync {

    public issue1(String name) {
        system = name;
    }

    @Test
    public void issue1() throws ExecutionException, InterruptedException {
        AsyncClient sclient = AsyncDatomic.clientPeerServer("k", "s", "localhost:8998");
        CompletableFuture<Either<CognitectAnomaly, AsyncConnection>> fcon = sclient.connect("hello");
        Either<CognitectAnomaly, AsyncConnection> either = fcon.get();
        Right right = (Right) either;
        AsyncConnection con = (AsyncConnection) right.right_value();
        AsyncDb sdb = con.db();
        CompletableFuture<Channel<Stream<?>>> qr = AsyncDatomic.q(
            map(
                Keyword.intern("query"), read("[:find ?movie-title :where [_ :movie/title ?movie-title]]"),
                Keyword.intern("args"), list(sdb.datomicDb())
            )
        );

        // chunk() works now...
        assertThat(
            //
            ((Right<?, Stream<?>>) qr.get().chunk()).right_value().toArray(),
            is(list(list("Commando"), list("The Goonies"), list("Repo Man")).toArray())
        );
    }
}

File: tests/src/test/scala-2.13+/datomicScala/client/issues/issue1.scala
//package datomicScala.client.issues
//
//import java.util.stream
//import java.util.stream.{Stream => jStream}
//import cats.effect.IO
//import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
//import datomic.Util
//import datomic.Util._
//import datomicClient.anomaly.{AnomalyWrapper, CognitectAnomaly, Forbidden, NotFound}
//import datomicScala.SpecAsync
//import datomicScala.client.api.async.AsyncDatomic
//import scala.concurrent.Future
//import scala.jdk.StreamConverters._
//
//
//class issue1 extends SpecAsync with AnomalyWrapper{
//
////  "create client" >> {
////    import clojure.lang.Keyword
////    import clojure.lang.PersistentHashMap
////    import com.sun.tools.javac.util.List
////    import datomic.Util
////    val sclient = AsyncDatomic.clientPeerServer("datomic", "datomic", "localhost:8998")
////    val fcon    = sclient.connect("dev-entity-system")
////    val either  = fcon.get
////    val right   = either.asInstanceOf[Nothing]
////    val con     = right.right_value.asInstanceOf[Nothing]
////    val sdb     = con.db
////    val qr      = AsyncDatomic.q(
////      PersistentHashMap.create(
////        Map.of(
////          Keyword.intern("query"),
////          Util.read("[:find ?roles :in $ :where [?e :LoadUser/roles2 " + ":LoadUserRole/R14][?e :LoadUser/roles2 ?roles]]"),
////          Keyword.intern("args"),
////          List.of(sdb.datomicDb)
////        )
////      ).asInstanceOf[PersistentHashMap]
////    )
////  }
//
//  "create client" >> {
//    import clojure.lang.Keyword
//    import clojure.lang.PersistentHashMap
//    import com.sun.tools.javac.util.List
//    import datomic.Util
//
//    val sclient = AsyncDatomic.clientPeerServer("datomic", "datomic", "localhost:8998")
//    val fcon    = sclient.connect("dev-entity-system")
//    val either  = fcon.get
//    val right   = either.asInstanceOf[Nothing]
//    val con     = right.right_value.asInstanceOf[Nothing]
//    val sdb     = con.db
//    val qr      = AsyncDatomic.q(
//      PersistentHashMap.create(
//        Map.of(
//          Keyword.intern("query"),
//          Util.read("[:find ?roles :in $ :where [?e :LoadUser/roles2 " + ":LoadUserRole/R14][?e :LoadUser/roles2 ?roles]]"),
//          Keyword.intern("args"),
//          List.of(sdb.datomicDb)
//        )
//      ).asInstanceOf[PersistentHashMap]
//    )
//  }
//
//}
